src/state-client.ts

```diff
@@ -284,157 +323,197 @@ class StateClient {
+  public validateCode = <ResponseType = unknown>(identifier: string, code: string) =>
+    this.signedRequest<{ pin: string }, ResponseType>(identifier, 'POST', 'validate_pin', { pin: code });
 
   /** @ignore */
-  public pinTokenExists = (identifier: string) => this.signedRequest(identifier, 'GET', 'pin_token_exists');
+  public pinTokenExists = <ResponseType = unknown>(identifier: string) => this.signedRequest<undefined, ResponseType>(identifier, 'GET', 'pin_token_exists');
 
   /** @ignore */
-  public revokeAllPINCodes = (identifier: string) => this.signedRequest(identifier, 'DELETE', 'pin');
+  public revokeAllPINCodes = <ResponseType = unknown>(identifier: string) => this.signedRequest<undefined, ResponseType>(identifier, 'DELETE', 'pin');
 
   /** @ignore */
-  public getEligibleCodes = (identifier: string) => this.signedRequest(identifier, 'GET', 'codes/eligible');
+  public getEligibleCodes = (identifier: string) => this.signedRequest<undefined, EligibleCodes>(identifier, 'GET', 'codes/eligible');
 
   /** @ignore */
-  public getRedeemedCodes = (identifier: string) => this.signedRequest(identifier, 'GET', 'codes/redeemed');
+  public getRedeemedCodes = (identifier: string) => this.signedRequest<undefined, RedeemedCodes>(identifier, 'GET', 'codes/redeemed');
 
   /** @ignore */
   public redeemCode = (identifier: string, prizeIndex: number) =>
-    this.signedRequest(identifier, 'POST', 'codes/redeem', JSON.stringify({ prize: prizeIndex }));
+    this.signedRequest<{ all_prizes: number }, RedeemResult>(identifier, 'POST', 'codes/redeem', { all_prizes: prizeIndex });
 
   /** @ignore */
   public getExtensionUsers = (identifier: string, cursor: string) =>
-    this.signedRequest(identifier, 'GET', `user_ids?cursor=${cursor || 0}`);
+    this.signedRequest<undefined, ExtensionUsersResult>(identifier, 'GET', `user_ids?cursor=${cursor || 0}`);
 
   /** @ignore */
   public joinExtensionTriviaTeam = (identifier: string) => this.signedRequest(identifier, 'POST', 'team_membership');
 
   /** @ignore */
   public getExtensionTriviaJoinedTeam = (identifier: string) =>
-    this.signedRequest(identifier, 'GET', 'team_membership');
+    this.signedRequest<undefined, TriviaTeam>(identifier, 'GET', 'team_membership');
 
   /** @ignore */
   public addExtensionTriviaQuestion = (identifier: string, triviaQuestion: TriviaQuestion) =>
-    this.signedRequest(identifier, 'POST', 'curated_poll_edit', JSON.stringify(triviaQuestion));
+    this.signedRequest<TriviaQuestion, undefined>(identifier, 'POST', 'curated_poll_edit', triviaQuestion);
 
   /** @ignore */
   public removeExtensionTriviaQuestion = (identifier: string, triviaQuestionID: string) =>
-    this.signedRequest(identifier, 'DELETE', 'curated_poll_edit', JSON.stringify({ id: triviaQuestionID }));
+    this.signedRequest<{ id: string }, undefined>(identifier, 'DELETE', 'curated_poll_edit', { id: triviaQuestionID });
```

Comment:
Do these endpoints actually return `undefined` or should these be `unknown` with a TODO to type correctly later?